assets/marketplace/oss-existingvpc.yaml

```diff
@@ -0,0 +1,1064 @@
+
+  HostedZoneID:
+    ConstraintDescription: ID of the hosted zone with Teleport domain
+    Description: ID of the hosted zone of teleport domain
+    Type: String
+
+  # Any existing VPC used must have both DNS support and DNS hostnames enabled
+  VPC:
+    Type: AWS::EC2::VPC::Id
+
+  # Needs to be set to the CIDR of the existing VPC
+  CIDRVPC:
+    Type: String
+
+  # Each provided subnet needs at least two available addresses
+  CIDRProxyA:
+    Type: String
+
+  CIDRProxyB:
+    Type: String
+
+  CIDRAuthA:
+    Type: String
+
+  CIDRAuthB:
+    Type: String
+
+  CIDRNodeA:
+    Type: String
+
+  CIDRNodeB:
+    Type: String
+
+Mappings:
+  AWSInstanceType2Arch:
+    t2.micro: {Arch: HVM64}
+
+  AWSRegionArch2AMI:
+    eu-west-1: {HVM64 : ami-b4d188cd}
+    us-east-1: {HVM64 : ami-2010b95d}
+    us-east-2: {HVM64 : ami-c50434a0}
+    us-west-2: {HVM64 : ami-e9741291}
+
+Resources:
+  # Internet gateway is a routing target for 
+  # internet routeable target
+  InternetGateway:
+    Type: AWS::EC2::InternetGateway
+    Properties:
+      Tags:
+      - Key: Cluster
+        Value: {Ref: DomainName}
+
+  # Attach this gateway to VPC
+  InternetGatewayAttachment:
+    Type: AWS::EC2::VPCGatewayAttachment
+    Properties:
+      InternetGatewayId: {Ref: InternetGateway}
+      VpcId: {Ref: VPC}
+
+  # VPC Routing table that sets up default
+  # route to use internet gateway,
+  # subnets using this table are public subnets
+  # from AWS perspective
+  PublicRoutingTable:
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: {Ref: VPC}
+      Tags:
+      - Key: Cluster
+        Value: {Ref: DomainName}
+      - Key: Type
+        Value: Public
+
+  # Public routing table routes all packets
+  # to the internet.
+  PublicRoute:
+    Type: AWS::EC2::Route
+    DependsOn: InternetGateway
+    Properties:
+      RouteTableId: {Ref: PublicRoutingTable}
+      DestinationCidrBlock: 0.0.0.0/0
+      GatewayId:
+        Ref: InternetGateway
+
+  # ProxySubnetA is a public subnet
+  # used by proxy in the first availability zone
+  ProxySubnetA:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId:
+        Ref: VPC
+      CidrBlock: {Ref: CIDRProxyA}
+      AvailabilityZone:
+        Fn::Select: [0, {'Fn::GetAZs': ''}]
+
+  ProxySubnetARouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      SubnetId: {Ref: ProxySubnetA}
+      RouteTableId: {Ref: PublicRoutingTable}
+
+  ProxySubnetB:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId:
+        Ref: VPC
+      CidrBlock: {Ref: CIDRProxyB}
+      AvailabilityZone:
+        Fn::Select: [1, {'Fn::GetAZs': ''}]
+
+  ProxySubnetBRouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      SubnetId: {Ref: ProxySubnetB}
+      RouteTableId: {Ref: PublicRoutingTable}
+
+  # Elastic IP in the first AZ
+  EIPA:
+    Type: AWS::EC2::EIP
+    Properties:
+      Domain: vpc
+
+  # NAT gateway in the first AZ
+  NATA:
```

Comment:
same for nat gateway - should require two nat gateways - one per AZ

Reply:
I don't think CloudFormation allows you to pass in the ID of an existing NAT Gateway to this function, it's expecting to create it itself and associate it with the ProxySubnet.

We could maybe ask for the IDs of ProxySubnetA and ProxySubnetB to be passed in as parameters rather than the CIDR block? I'm not sure how we validate that they are public rather than private subnets if we don't create them, though.